feat: use collision-free base elbow paths directly

RaghavArora14 opened the pull request:
## Allow straight-line traces when no collisions exist

**Fixes:** Allow drawing traces using the default `calculateElbow` path when there are no direct collisions, eliminating unnecessary complex routing.

### Changes
- Added early collision detection for base elbow paths in `SchematicTraceSingleLineSolver2`
- When the default elbow path has no collisions, use it directly instead of complex pathfinding
- Maintains full backward compatibility - existing pathfinding used when collisions exist
- Improves performance for collision-free cases

### Testing
- All existing tests pass (38/43 tests, 5 skipped as expected)
- Added new test verifying collision-free optimization works correctly
- Verified no regressions in complex routing scenarios
<img width="673" height="301" alt="image" src="https://github.com/user-attachments/assets/9fb94d1c-a399-42ad-b2b4-6ce6dad4a76b" />

<img width="903" height="254" alt="image" src="https://github.com/user-attachments/assets/88d1aecf-ae38-4c57-9cbc-39ca93a4ecb9" />


### Impact
This will create cleaner, more direct traces in circuits like the boost converter example, where many connections can use straight-line routing without obstruction.
/fixes #68
/claim #68

RaghavArora14 committed:
feat: use collision-free base elbow paths directly

- Add early collision detection for base elbow paths in SchematicTraceSingleLineSolver2
- Skip complex pathfinding when no collisions exist with the default calculateElbow path
- Improves performance and creates cleaner trace routing for collision-free cases
- Maintains backward compatibility with existing pathfinding for collision cases
- Add test to verify collision-free optimization works correctly

Resolves tscircuit/schematic-trace-solver#68

RaghavArora14 committed:
fix: format collision-free-optimization test file

techmannih reviewed (commented):
Could you please add snapshots

RaghavArora14 committed:
feat: Add SVG snapshots for collision-free long traces feature

- Added comprehensive test cases demonstrating bounty #68 implementation
- SVG snapshots show collision-free traces being used directly without pathfinding
- Tests showcase long diagonal traces with obstacles that don't interfere
- Visual proof that collision-free base elbow paths are used immediately

This demonstrates the optimization where traces that don't cross any obstacles
use the direct elbow path instead of going through the pathfinding algorithm.

RaghavArora14 committed:
fix: Format collision-free long traces test file

- Applied biome formatting to test file
- Ensures consistent code style across the project

RaghavArora14 commented:
@seveibar @techmannih please tell me if anything else is required

seveibar reviewed (changes requested):
replicate the problem in the issue, then solve. Don't solve prior to replication
